commit 0db63d6cf772ce2d585c8bb8a683636f28086ce5
Author: Kristof Provost <kristof@codepro.be>
Date:   Sat Aug 25 14:12:51 2018 +0200

    pfsync: Ensure uninit is done before pf
    
    pfsync touches pf memory (for pf_state and the pfsync callback
    pointers), not the other way around. We need to ensure that pfsync is
    torn down before pf.

diff --git sys/netpfil/pf/if_pfsync.c sys/netpfil/pf/if_pfsync.c
index c45354e0a42..7efa089ec14 100644
--- sys/netpfil/pf/if_pfsync.c
+++ sys/netpfil/pf/if_pfsync.c
@@ -2625,11 +2625,8 @@ vnet_pfsync_uninit(const void *unused __unused)
 	swi_remove(V_pfsync_swi_cookie);
 	swi_remove(V_pfsync_msg_swi_cookie);
 }
-/*
- * Detach after pf is gone; otherwise we might touch pfsync memory
- * from within pf after freeing pfsync.
- */
-VNET_SYSUNINIT(vnet_pfsync_uninit, SI_SUB_INIT_IF, SI_ORDER_SECOND,
+
+VNET_SYSUNINIT(vnet_pfsync_uninit, SI_SUB_PROTO_FIREWALL, SI_ORDER_FOURTH,
     vnet_pfsync_uninit, NULL);
 
 static int
